Explain disposables and memory risky objects. (#10039)
flutter/website: src/tools/devtools/memory.md

[Dart garbage collection]: {{site.medium}}/flutter/flutter-dont-fear-the-garbage-collector-d69b3ff1ca30

### Object types

#### Disposable object

> A Dart object, that has method `dispose()`. The method, by convention, is expected to be invoked
> when the object isn't needed anymore to avoid memory leaks.

Reviewer: ```suggestion
A disposable object is any Dart object that defines a `dispose()` method.
To avoid memory leaks, Flutter expects that `dispose` is invoked when the
object isn't needed anymore.
```

> when the object is not needed any more, to avoid memory leaks.

Reviewer 2: wording/ punctuation nit: "A Dart object that has a method `dispose()`, which, by convention, is expected to be invoked when the object is not needed anymore to avoid memory leaks."

> A Dart object, that has method `dispose()`, which, by de-facto convention, is expected to be invoked,
> when the object is not needed any more, to avoid memory leaks.

Reviewer: ```suggestion
A Dart object that has a `dispose()` method is, by convention, expected to be invoked
when the object isn't needed anymore to avoid memory leaks.
```

Author: Used different rephrasing to keep meaning.

#### Memory-risky object

> A Dart object that, if not disposed or disposed but not GCed, might cause a memory leak.

Reviewer: ```suggestion
A Dart object if not disposed or disposed but not GCed, might cause a memory leak.
```

> A Dart object, that, if not disposed or disposed but not GCed, may cause a memory leak.

Reviewer 2: nit: remove comma after object

> A Dart object, that, if not disposed or disposed but not GCed, may cause a memory leak.

Reviewer: Can we even remove "might"? Doesn't it for sure cause a memory leak?

```suggestion
A Dart object that is not disposed, or is disposed but not GCed, might cause a memory leak.
```

Author: Nope. It causes memory leak if only not disposed or not GCed.
It is ok, if it is disposed and GCed.
